### DISCUSSION TITLE
What's the point of fragment?

### DISCUSSION THREAD

**Discussion by inwardmovement** [withastro/docs#722] [Ideas]
I don't understand [fragments](https://docs.astro.build/en/core-concepts/astro-components/#fragments--multiple-elements) as the output is the same with or without wrapping in `<>...</>`. Can you tell me?

This:
```
<ul>
  {items.map((item) => (
    <>
      <li>Red {item}</li>
      <li>Blue {item}</li>
      <li>Green {item}</li>
    </>
  ))}
</ul>
```

Or this:
```
<ul>
  {items.map((item) => (
    <li>Red {item}</li>
    <li>Blue {item}</li>
    <li>Green {item}</li>
  ))}
</ul>
```

Both outputs to:
```
<ul>
  <li>Red Dog</li>
  <li>Blue Dog</li>
  <li>Green Dog</li>
  <li>Red Cat</li>
  <li>Blue Cat</li>
  <li>Green Cat</li>
  <li>Red Platypus</li>
  <li>Blue Platypus</li>
  <li>Green Platypus</li>
</ul>
```

**Comment by delucis** [comment]
One useful thing is that you can `set:html` on a fragment:

```astro
<Fragment set:html={htmlString} />
```

You can also do that on an HTML element, but fragment lets you inject just the HTML with no wrapper element.

Anyone else have any favourite uses?

**Reply by inwardmovement** [reply]
Thanks for the tip!
So maybe we should not use "must" in "you **must** wrap these multiple elements inside of a Fragment", right?
And instead provide some tips fragments are useful for.

**Reply by delucis** [reply]
Maybe! Sounds related to https://github.com/withastro/docs/issues/369

**Comment by ArmandPhilippot** [comment]
We're closing stale and outdated discussion to keep only the discussions still relevant to the current project version and/or the current docs state.

This seems [resolved to me](https://docs.astro.build/en/reference/astro-syntax/#fragments).

If you think this is a mistake and that this discussion should remain open, please feel free to ping me, I'll take a second look at it.

Thank your for your contribution!
